File: solutions/python/protein-translation/1/protein_translation.py
CODONS = {
    "Methionine": ["AUG"],
    "Phenylalanine": ["UUU", "UUC"],
    "Leucine": ["UUA", "UUG"],
    "Serine": ["UCU", "UCC", "UCA", "UCG"],
    "Tyrosine": ["UAU", "UAC"],
    "Cysteine": ["UGU", "UGC"],
    "Tryptophan": ["UGG"],
    "STOP": ["UAA", "UAG", "UGA"]
}

def proteins(strand):
    result = []
    for i in range(0, len(strand), 3):
        input_codon = strand[i:i+3]
        if input_codon in CODONS["STOP"]:
            return result
        for amino, codons in CODONS.items():
            if input_codon in codons:
                result.append(amino)
    return result

File: solutions/csharp/protein-translation/1/ProteinTranslation.cs
public static class ProteinTranslation {
  private static readonly Dictionary<string, string> CodonToProtein = new Dictionary<string, string>() {
    {"AUG", "Methionine"},
    {"UUU", "Phenylalanine"},
    {"UUC", "Phenylalanine"},
    {"UUA", "Leucine"},
    {"UUG", "Leucine"},
    {"UCU", "Serine"},
    {"UCC", "Serine"},
    {"UCA", "Serine"},
    {"UCG", "Serine"},
    {"UAU", "Tyrosine"},
    {"UAC", "Tyrosine"},
    {"UGU", "Cysteine"},
    {"UGC", "Cysteine"},
    {"UGG", "Tryptophan"},
    {"UAA", "STOP"},
    {"UAG", "STOP"}
  };

  public static string[] Proteins(string strand) => Enumerable.Range(0, strand.Length / 3)
      .Select(i => strand.Substring(i * 3, 3))
      .Select(codon => CodonToProtein.ContainsKey(codon) ? CodonToProtein[codon] : null)
      .TakeWhile(protein => protein != "STOP")
      .Where(protein => protein != null)
      .ToArray();
}